From 15873d368898c7f4f7d7673464ad59135024ea3b Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Fri, 7 Nov 2014 14:48:21 +0100
Subject: [PATCH 1/2] Compatibility with Maven 3.4.0

This adds compatibility with latest Maven APIs by implementing new
interface methods introduced in Maven 3.4.0 (MNG-5663 and others).

Signed-off-by: Mikolaj Izdebski <mizdebsk@redhat.com>
---
 .../aether/internal/ant/AntModelResolver.java      | 22 +++++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/src/main/java/org/eclipse/aether/internal/ant/AntModelResolver.java b/src/main/java/org/eclipse/aether/internal/ant/AntModelResolver.java
index dc7ef72..a177b17 100644
--- a/src/main/java/org/eclipse/aether/internal/ant/AntModelResolver.java
+++ b/src/main/java/org/eclipse/aether/internal/ant/AntModelResolver.java
@@ -16,6 +16,8 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Parent;
 import org.apache.maven.model.Repository;
 import org.apache.maven.model.building.FileModelSource;
 import org.apache.maven.model.building.ModelSource;
@@ -77,7 +79,13 @@ class AntModelResolver
     public void addRepository( Repository repository )
         throws InvalidRepositoryException
     {
-        if ( !repositoryIds.add( repository.getId() ) )
+        addRepository( repository, false );
+    }
+
+    public void addRepository( Repository repository, boolean replace )
+        throws InvalidRepositoryException
+    {
+        if ( !repositoryIds.add( repository.getId() ) && !replace )
         {
             return;
         }
@@ -145,4 +153,16 @@ class AntModelResolver
         return new FileModelSource( pomFile );
     }
 
+    public ModelSource resolveModel( Parent parent )
+        throws UnresolvableModelException
+    {
+        return resolveModel( parent.getGroupId(), parent.getArtifactId(), parent.getVersion() );
+    }
+
+    public ModelSource resolveModel( Dependency dependency )
+        throws UnresolvableModelException
+    {
+        return resolveModel( dependency.getGroupId(), dependency.getArtifactId(), dependency.getVersion() );
+    }
+
 }
-- 
2.7.4

